src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright 2019 IBM All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Assertation 1: `type` is a necessary property and has four possible values: `apiKey`, `HTTP`, `oauth2`, `openIdConnect`
+// Assertation 2: name property is required for `apiKey` type
+// Assertation 3: `in` property is required for `apiKey` type, valid values are: `query`, `header` or `cookie`
+// Assertation 4: `scheme` property` is required for `http` type
+// Assertation 5: `flows` object is required for `oauth2` type
+// Assertation 6: `opedIdConnectUrl` property is required for `openIdConnect` and must be a valid url
+
+const validator = require('validator');
+
+module.exports.validate = function({ resolvedSpec }) {
+  const API_KEY = 'apiKey';
+  const OAUTH2 = 'oauth2';
+  const HTTP = 'http';
+  const OPENID_CONNECT = 'openIdConnect';
+  const authTypes = [API_KEY, HTTP, OAUTH2, OPENID_CONNECT];
+  const errors = [];
+  const warnings = [];
+  const securitySchemes =
+    resolvedSpec.components && resolvedSpec.components.securitySchemes;
+
+  for (const key in securitySchemes) {
+    const path = `securitySchemes.${key}`;
+    const security = securitySchemes[key];
+    const type = security.type;
+
+    if (!type) {
+      errors.push({
+        message: 'security scheme is missing required field `type`',
+        path
+      });
+    } else if (authTypes.indexOf(type) === -1) {
+      errors.push({
+        message:
+          '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`',
+        path
```

Comment:
I think the `path` should be as specific as possible, so in this case we should point directly at the `type` property within the securityScheme.
```suggestion
        path+'.type'
```